src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -57,17 +57,27 @@ static void HandleNodeResolve(void * context, DnssdService * result, CHIP_ERROR
     Platform::CopyString(nodeData.hostName, result->mHostName);
     Platform::CopyString(nodeData.instanceName, result->mName);
 
+    size_t addressesFound = 0;
     if (result->mAddress.HasValue())
```

Comment:
Would there be value to not split first + rest addresses? Why not have the data  in result and then 'all IPs inside the span'?

Reply:
More churn for existing consumers, mostly.  Filed https://github.com/project-chip/connectedhomeip/issues/15279 to do this as a followup if we want.